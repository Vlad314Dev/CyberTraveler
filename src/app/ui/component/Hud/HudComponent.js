import './HudStyle';

import React, { PureComponent } from 'react';
import HealthBar from 'UIComponent/HealthBar';
import Score from 'UIComponent/Score';
import Weapons from 'UIComponent/Weapons';

class HudComponent extends PureComponent
{
    render()
    {
        return (
            <div block="Hud">
                <HealthBar/>
                <Weapons/>
                <Score/>
            </div>
        );
    }
}

export default HudComponent;
