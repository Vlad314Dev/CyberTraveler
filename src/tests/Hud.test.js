/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import HudComponent from '../app/ui/component/Hud/HudComponent';

describe('Hud', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <HudComponent/>
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <HudComponent/>
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
