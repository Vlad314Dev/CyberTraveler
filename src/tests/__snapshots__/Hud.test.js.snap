// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hud Should render layout 1`] = `
Array [
  <div
    className="Hud"
  >
    <Memo(Connect(HealthBarContainer)) />
    <Memo(Connect(WeaponsContainer)) />
    <Memo(Connect(ScoreContainer)) />
  </div>,
]
`;
